### ISSUE TITLE
Frozen string failure in output

### ISSUE THREAD

**Issue by viraptor** [cucumber/cucumber-ruby#1295]
## Summary

A random run ended up with:
```
can't modify frozen String (RuntimeError)
/project/path/vendor/bundle/ruby/2.3.0/gems/cucumber-3.1.0/lib/cucumber/formatter/html.rb:571:in `snippet'
/project/path/vendor/bundle/ruby/2.3.0/gems/cucumber-3.1.0/lib/cucumber/formatter/html.rb:293:in `extra_failure_content'
...
```

## Possible Solution

This error doesn't seem to make any sense.
- if the string is appended there, the converter is NullConverter
- in that case `highlighted == raw_code`
- `raw_code` comes from `snippet_for`
- `snippet_for` returns an array

Unless something, somewhere overrides the behaviour, this can't happen 🤔But this project doesn't override anything in related classes.

## Steps to Reproduce (for bugs)

Unfortunately the ordering is random and I can't reproduce it.

## Your Environment

* Version used: 3.1.0
* Operating System and version: Ubuntu trusty, ruby 2.3
* Link to your project:

**Comment by viraptor** [comment]
No, I don't have the Syntax gem

**Comment by xtrasimplicity** [comment]
@viraptor I've created [a PR](https://github.com/cucumber/cucumber-ruby/pull/1296) with the intention of preparing a new release in the next few weeks.

In the meantime, I would recommend either temporarily monkey-patching `Cucumber::Formatter::Html.snippet` as per https://github.com/cucumber/cucumber-ruby/pull/1287 or installing the `Syntax` gem. Once we've released a new version of `cucumber-ruby`, you will be able to remove the monkey patch or uninstall the `Syntax` gem. :)
